proto/src/rr/dns_class.rs

```diff
@@ -133,7 +133,8 @@ impl From<DNSClass> for u16 {
             DNSClass::HS => 4,
             DNSClass::NONE => 254,
             DNSClass::ANY => 255,
-            DNSClass::OPT(version) => version,
+            // see https://tools.ietf.org/html/rfc6891#section-6.1.2
+            DNSClass::OPT(max_payload_len) => max_payload_len,
```

Comment:
IIUC, here you should reject any value less than less than or equal to 255. In particular, you must not allow `max_payload_len` to be in {1, 3, 4, 254, 255} since then the conversion would be ambiguous.

Reply:
Yeah. I’ve been thinking of a better fix here since writing this. It actually occurs to me that anything less than 512 would be out of bounds. I’ll update.